### PR TITLE
Improve walkeeper replication error messages & context

### PR DESCRIPTION
**Pull request by sharnoff**
Just a pass through the all of the errors generated in the `walkeeper::replication` module to try to improve them a little bit where I can. Some of this is pretty subjective, would love feedback from anyone & everyone if there is any.

### CODE REVIEW

File: walkeeper/src/replication.rs [hunk 1]
```diff
@@ -120,13 +121,12 @@ impl ReplicationConn {
...
         // If that failed, try it without the .partial extension.
-        match File::open(&wal_file_path) {
-            Ok(opened_file) => Ok(opened_file),
-            Err(e) => {
-                error!("Failed to open log file {:?}: {}", &wal_file_path, e);
-                Err(e.into())
-            }
-        }
+        File::open(&wal_file_path)
+            .with_context(|| format!("Failed to open log file {:?}", wal_file_path))
```
Comment (hlinnaka) [inline walkeeper/src/replication.rs:125]: "Log file" sounds like non-critical debugging information, but failure to open the WAL file is serious. I'd suggest "Failed to open WAL file".

File: walkeeper/src/replication.rs [hunk 2]
```diff
@@ -216,16 +216,18 @@ impl ReplicationConn {
...
 
             // Write some data to the network socket.
             pgb.write_message(&BeMessage::XLogData(XLogDataBody {
                 wal_start: start_pos.0,
                 wal_end: end_pos.0,
                 timestamp: get_current_timestamp(),
                 data: &file_buf,
-            }))?;
+            }))
+            .context("Failed to send XLogData")?;
```
Comment (hlinnaka) [inline walkeeper/src/replication.rs:230]: These two errors are quite different: The first indicates that something is pretty badly wrong: the point of a safekeeper is to not lose the WAL, so if it it cannot read the WAL file, that's serious. In contrast, the second error can happen whenever the network connection is broken, which is more or less normal.

I think this is OK for now, a message is nicer than no message, which is what we had before this PR. But this is a good example to consider in https://github.com/zenithdb/zenith/issues/551.

**Review by hlinnaka — APPROVED** [review]
Nice cleanup!

File: walkeeper/src/replication.rs (hunk 1 above)
Comment (sharnoff) [reply walkeeper/src/replication.rs]: Fixed :)

File: walkeeper/src/replication.rs (hunk 2 above)
Comment (sharnoff) [reply walkeeper/src/replication.rs:230]: After thinking about this for a bit: is there any reason here that we shouldn't just panic, if things are clearly pretty bad?

And if we shouldn't panic, what would the optimal behavior be?

**Pull request merged**